File: DAY 21/ReverseWordsInAGivenString.cpp
#include <bits/stdc++.h>
using namespace std;

void revWords(string s){
    vector<string> tmp;
    string str = " ";
    for(unsigned int i=0;i<s.length();i++){
        if(s[i]==' '){
            tmp.push_back(str);
            str = "";
        }
        else str+=s[i];
    }
    tmp.push_back(str);
    int i;
    for(i=tmp.size()-1;i>0;i--) cout << tmp[i] << " ";
    cout << tmp[0] << endl;
}

int main(){
    string s = "this is just a test";
    revWords(s);
    return 0;
}

File: DAY 21/ReverseWordsInAGivenString.java
public class ReverseWordsInAGivenString {
    static void rev(char str[], int start, int end){
        for(char temp; start<=end;start++,end--){
            temp = str[start];
            str[start] = str[end];
            str[end] = temp;
        }
    }
    static char[] revWords(char[] s){
        int start = 0;
        for(int end=0;end<s.length;end++){
            if(s[end]==' '){
                rev(s, start, end);
                start = end + 1;
            }
        }
        rev(s, start, s.length-1);
        rev(s, 0, s.length-1);
        return s;
    }
    public static void main(String[] args){
        String s = "this is just a test";
        char[] p = revWords(s.toCharArray());
        System.out.print(p);
    }
}
